### DISCUSSION TITLE
Sending Email from Save Handler

### DISCUSSION THREAD

**Discussion by TheWebbe** [serenity-is/Serenity#6349] [Help]
Hi All, 
I am looking to send an email (based on certain criteria) after a record is saved. It seems most of the topics here are related to the old methods of sending email (either by creating a SMTP client manually or using the emailhelper method which doesn't exist any more). I am currently running the latest version (6.0.0).

The only reference I had was the forgot password functionality, though I'm not sure that will relate to this as I was intending. 

My thought was to create a new class in the save handler to send the email notification: 

```
public void sendNotification(MyRow Row,[FromServices] IEmailSender emailSender, [FromServices] IOptions<EnvironmentSettings> options = null)
        {
            
            var emailSubject = "Test Email Subject";
            var emailBody = "Test Email Body";
            if (emailSender is null)
                throw new ArgumentNullException(nameof(emailSender));

            emailSender.Send(subject: emailSubject, body: emailBody, mailTo: Row.UserEmail);
        }
```

then calling it from the AfterSave() function. 

The issue is, I can't call that function without passing it emailSender. Is there not a global EmailSender that I can pull in, or do I need to create a new one for this task. If I need to create a new one, is there an example somewhere that shows how to configure this correctly? 

Thanks!

**Comment by Zahar661** [comment]
Were you able to send the letter?

**Reply by Zahar661** [reply]
![image](https://user-images.githubusercontent.com/26430136/197943569-a954c1ad-7b56-4b4b-bc24-7b0426a7a9eb.png)
![image](https://user-images.githubusercontent.com/26430136/197943774-e1bf4db9-d5ae-44aa-aadb-078a897c2a0b.png)
![image](https://user-images.githubusercontent.com/26430136/197943890-bf9fe57d-b0d7-4f1c-a05f-d23cca9c524e.png)

**Reply by cp1319421** [reply]
what's need update in version 6.7.6?
![PIC1](https://github.com/serenity-is/Serenity/assets/19610257/0adad659-577d-4137-9436-1c15ee49dfa0)
![PIC2](https://github.com/serenity-is/Serenity/assets/19610257/185b1e11-edcc-4a17-a737-e3468b9594c0)
